tests/TestData/Providers/StubsTestDataProviders.php

```diff
@@ -47,32 +45,151 @@ public static function stubClassProvider(): ?Generator
         }
     }
 
-    public static function coreStubMethodProvider(): ?Generator
+    public static function stubMethodProvider(): ?Generator
     {
-        foreach (PhpStormStubsSingleton::getPhpStormStubs()->getCoreClasses() as $className => $class) {
+        $classesAndInterfaces = PhpStormStubsSingleton::getPhpStormStubs()->getClasses() +
+            PhpStormStubsSingleton::getPhpStormStubs()->getInterfaces();
+        foreach ($classesAndInterfaces as $className => $class) {
             foreach ($class->methods as $methodName => $method) {
                 yield "method {$className}::{$methodName}" => [$methodName, $method];
             }
         }
+    }
+
+    public static function stubMethodParametersWithoutScalarTypeHintsProvider(): ?Generator
+    {
+        foreach (EntitiesFilter::getFiltered(PhpStormStubsSingleton::getPhpStormStubs()->getCoreClasses()) as $className => $class) {
+            foreach (EntitiesFilter::getFilteredFunctions($class) as $methodName => $method) {
+                $firstSinceVersion = StubsTypeHintsTest::getFirstSinceVersion($method);
+                if ($class !== null && !$method->isFinal && !$class->isFinal && $firstSinceVersion !== -1 && $firstSinceVersion < 7) {
+                    foreach (EntitiesFilter::getFilteredParameters($method, StubProblemType::PARAMETER_HAS_SCALAR_TYPEHINT) as $parameter) {
+                        yield "method {$className}::{$methodName}($parameter->name)" => [$method, $parameter];
+                    }
+                }
+            }
+        }
 
-        foreach (PhpStormStubsSingleton::getPhpStormStubs()->getCoreInterfaces() as $interfaceName => $interface) {
-            foreach ($interface->methods as $methodName => $method) {
-                yield "method {$interfaceName}::{$methodName}" => [$methodName, $method];
+        foreach (EntitiesFilter::getFiltered(PhpStormStubsSingleton::getPhpStormStubs()->getCoreInterfaces()) as $interfaceName => $interface) {
```

Comment:
Let's add `isFinal = false`  to all interfaces and interfaces' methods and combine both foreaches.